server/src/server.ts

```diff
@@ -791,17 +792,27 @@ function getCompileCommand(settings: CairoLSSettings, tempFiles: TempFiles, text
 	let cairoPathParam = "";
 
 	const sourceDir = settings.sourceDir;
+	const cairoPath = settings.cairoPath;
 
-	if (workspaceFolders.length > 0) {
+	if (workspaceFolders.length > 0 || cairoPath.length > 0) {
 		cairoPathParam = '--cairo_path=';
-		for (i = 0; i < workspaceFolders.length; i++) {
-			cairoPathParam += appendSourceDir(workspaceFolders[i], sourceDir);
+
+		for (i = 0; i < cairoPath.length; i++) {
+			cairoPathParam += cairoPath[i]
+
 			if (i < workspaceFolders.length - 1) {
 				cairoPathParam += ':';
 			}
 		}
+
+		for (i = 0; i < workspaceFolders.length; i++) {
+			cairoPathParam += ':';
```

Comment:
If `cairoPath` was empty, this appends `:` at the beginning which isn't correct.

Reply:
done